src/dawn/Optimizer/AccessComputation.cpp

```diff
@@ -256,6 +256,7 @@ class AccessMapper : public iir::ASTVisitor {
     // the block statements as the if/then/else block of the if-statement has been collapsed into
     // one single
     // vector of block statements
+    appendNewAccesses();
```

Comment:
This breaks gtclang.

Reply:
I am wondering why gtclang was green. I pushed a fix (I hope), but I admit that I don't fully understand how if statements and block statements interact. But block statements are broken in gtclang anyway, so I only fix them for iir as far as they compile and run now.